diff --git a/Dockerfile b/Dockerfile
index aaaaaaaaaa..bbbbbbbbbb 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -28,6 +28,11 @@ COPY . ./
 ARG NODE_ENV=production
 
 RUN git submodule update --init
+# feat/nanoid-filename
+RUN wget -O - https://github.com/outloudvi/misskey/commit/d56003fecf990bed379d67f9ef7b93973250695e.patch | git apply && \
+# feat/i-5964
+    wget -O - https://github.com/outloudvi/misskey/commit/39b4b28f647d57e4819d793ad6303dd9348deae2.patch | git apply && \
+    pnpm install
 RUN pnpm build
 
 FROM node:-slim AS runner
