diff --git a/Dockerfile b/Dockerfile
index aaaaaaaaa..bbbbbbbbb 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -18,6 +18,8 @@ RUN corepack enable
 
 WORKDIR /misskey
 
+RUN wget -O - https://github.com/outloudvi/misskey/commit/6c77dfdacf43f36829877527161f4b9f59260b29.patch | git apply
+
 COPY --link ["pnpm-lock.yaml", "pnpm-workspace.yaml", "package.json", "./"]
 COPY --link ["scripts", "./scripts"]
 COPY --link ["packages/backend/package.json", "./packages/backend/"]
@@ -35,6 +37,7 @@ RUN --mount=type=cache,target=/root/.local/share/pnpm/store,sharing=locked  COPY --link . ./
 
 RUN git submodule update --init
+RUN wget -O - https://github.com/outloudvi/misskey/commit/b27010bada3ff6cb077b97a5b1fd7367fd66a5e7.patch | git apply
 RUN pnpm build
 RUN rm -rf .git/
