diff --git a/Dockerfile b/Dockerfile
index 3876b5f6ce..e499a6565c 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -28,6 +28,9 @@ COPY . ./
 ARG NODE_ENV=production
 
 RUN git submodule update --init
+# feat/nanoid-filename
+RUN wget -O - https://github.com/outloudvi/misskey/commit/57beb79ca1a83222632a486086983f06f4bbb606.patch | git apply && \
+    pnpm install
 RUN pnpm build
 
 FROM node:${NODE_VERSION}-slim AS runner
