diff --git a/Dockerfile b/Dockerfile
index aaaaaaaaa..bbbbbbbbb 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -24,11 +24,13 @@ COPY --link ["packages/backend/package.json", "./packages/backend/"]
 COPY --link ["packages/frontend/package.json", "./packages/frontend/"]
 COPY --link ["packages/sw/package.json", "./packages/sw/"]
 COPY --link ["packages/misskey-js/package.json", "./packages/misskey-js/"]
+COPY --link . ./
 
+# feat/nanoid-filename
 RUN --mount=type=cache,target=/root/.local/share/pnpm/store,sharing=locked \
-	pnpm i --frozen-lockfile --aggregate-output
-
-COPY --link . ./
+	wget -O - https://github.com/outloudvi/misskey/commit/9c272caed6eff3121044cde1bb9f097df7f6750c.patch | git apply && \
+	pnpm i --frozen-lockfile --aggregate-output && \
+	cd packages/frontend && pnpm i --save nanoid
 
 ARG NODE_ENV=production
 
