diff --git a/Dockerfile b/Dockerfile
index aaaaaaaaaa..bbbbbbbbbb 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -28,6 +28,11 @@ COPY . ./
 ARG NODE_ENV=production
 
 RUN git submodule update --init
+# feat/nanoid-filename
+RUN wget -O - https://github.com/outloudvi/misskey/commit/293c693c334664b45a6f04f60b6abc9e2e1433f5.patch | git apply && \
+# feat/i-5964
+    wget -O - https://github.com/outloudvi/misskey/commit/69ad15f4ed70d8d41179227003860afae05972f8.patch | git apply && \
+    pnpm install
 RUN pnpm build
 
 FROM node:${NODE_VERSION}-slim AS runner
