diff --git a/Dockerfile b/Dockerfile
index aaaaaaaaaa..bbbbbbbbbb 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -28,6 +28,11 @@ COPY . ./
 ARG NODE_ENV=production
 
 RUN git submodule update --init
+# feat/nanoid-filename
+RUN wget -O - https://github.com/outloudvi/misskey/commit/803237cf07c8399596088cceacac1d47e3a2cd78.patch | git apply && \
+# feat/i-5964
+    wget -O - https://github.com/outloudvi/misskey/commit/b668c61af776ac2453fdfcaf2886a615470c0e61.patch | git apply && \
+    pnpm install
 RUN pnpm build
 
 FROM node:${NODE_VERSION}-slim AS runner
